### PR TITLE
nnp_training updated

### PR DESCRIPTION
**Pull request by farhadrgh**
- Weight initialization to Kaiming normal with a=1
- Biases initialized to zeros
- exponential loss is removed from the training process
- The pretraining loop is removed

### CODE REVIEW

File: examples/nnp_training.py [hunk 1]
```diff
@@ -218,19 +239,27 @@
...
 ###############################################################################
-# The way ANI trains a neural network potential looks like this:
-#
-# Phase 1: Pretrain the model by minimizing MSE loss
-#
-# Phase 2: Train the model by minimizing the exponential loss, until validation
-# RMSE no longer improves for a certain steps, decay the learning rate and repeat
-# the same process, stop until the learning rate is smaller than a certain number.
+# Setting up a learning rate scheduler to do learning rate decay
+scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer, factor=0.5, patience=100)
```
Comment (zasdfgbnm) [inline examples/nnp_training.py]: `ReduceLROnPlateau` should have a `threshold=0`

File: examples/nnp_training.py [hunk 2]
```diff
@@ -218,19 +239,27 @@
...
+
+###############################################################################
+# Train the model by minimizing the mase loss, until validation RMSE no longer
```
Comment (zasdfgbnm) [inline examples/nnp_training.py]: nit: MSE

File: examples/nnp_training.py [hunk 3]
```diff
@@ -218,19 +239,27 @@
...
-pretrained = os.path.isfile(latest_checkpoint)
+
+###############################################################################
+# Resume training from previously saved checkpoints:
+if os.path.isfile(latest_checkpoint):
+    checkpoint = torch.load(latest_checkpoint)
+    model.load_state_dict(checkpoint['nn'])
+    optimizer.load_state_dict(checkpoint['optimizer'])
+
+    if 'scheduler' in checkpoint:
```
Comment (zasdfgbnm) [inline examples/nnp_training.py]: Since there is no more pretraining, we are always saving checkpoints for scheduler, so this if now is always true.

File: examples/nnp_training.py [hunk 4]
```diff
@@ -156,6 +156,27 @@
...
+#   https://pytorch.org/docs/stable/_modules/torch/nn/modules/linear.html#Linear
+
+
+def init_params(m):
+    if isinstance(m, torch.nn.Linear):
+        torch.nn.init.kaiming_normal_(m.weight, a=math.sqrt(5))
+        torch.nn.init.zeros_(m.bias)
+
+
+nn.apply(init_params)
```
Comment (zasdfgbnm) [inline examples/nnp_training.py]: Great! very clean!

Comment (farhadrgh) [reply examples/nnp_training.py]: Great! thanks for comments, I applied the changes!

File: examples/nnp_training.py [hunk 5]
```diff
@@ -156,6 +156,27 @@
...
+#   We initialize the weights similarly but from the normal distribution.
+#   The biases were initialized to zero.
+#
+# .. _TORCH.NN.MODULES.LINEAR:
+#   https://pytorch.org/docs/stable/_modules/torch/nn/modules/linear.html#Linear
+
+
+def init_params(m):
+    if isinstance(m, torch.nn.Linear):
+        torch.nn.init.kaiming_normal_(m.weight, a=math.sqrt(5))
```
Comment (zasdfgbnm) [inline examples/nnp_training.py]: Wait, should this be a=1? The equation in https://pytorch.org/docs/stable/nn.html#torch.nn.init.kaiming_normal_ is
std=sqrt(2/(1+a^2)fan_in)...

Comment (farhadrgh) [reply examples/nnp_training.py]: yes, you're right! Kaiming uniform and normal are slightly different!
